File: June Challenge/Week2/Java/Day_9/IsSubsequence.java
//Problem:https://leetcode.com/problems/is-subsequence/
//Solution:
class Solution {
    public boolean isSubsequence(String s, String t) {
        if(s.length()==0){
            return true;
        }
        char a=s.charAt(0);
        for(int i=0;i<t.length();i++){
            if(a==t.charAt(i)){
                s=s.substring(1);
                t=t.substring(i+1);
                return isSubsequence(s,t);
            }
        }
        return false;
    }
}

File: June Challenge/Week2/C++/Day_9/isSubsequence.cpp

class Solution {
public:
    bool isSubsequence(string s, string t) {

        int m= s.length()+1;
        int n= t.length() +1;

        vector<vector<int>> A;
        for(int i=0;i<m;i++){
            vector<int> v(n,0);
            A.push_back(v);
        }

        for(int i=1;i<m;i++){
            for(int j=1;j<n;j++){
                if(t[j-1]==s[i-1]){
                    A[i][j]=A[i-1][j-1]+1;
                }else{
                    A[i][j]=max(A[i-1][j],A[i][j-1]);
                }
            }
        }
        if(A[m-1][n-1]==s.length()){
            return true;
        }else{
            return false;
        }
    }
};
